common/src/node.rs

```diff
@@ -0,0 +1,59 @@
+/*the name of this module is pretty arbitrary at the moment*/
+
+use crate::graph::Graph;
+use crate::world::Material;
+use crate::worldgen::NodeState;
+use crate::Chunks;
+
+pub type DualGraph = Graph<Node>;
+
+pub struct Node {
+    pub state: NodeState,
+    /// We can only populate chunks which lie within a cube of populated nodes, so nodes on the edge
+    /// of the graph always have some `None` chunks.
+    pub chunks: Chunks<Chunk>,
+}
+
+#[derive(PartialEq)]
+pub enum VoxelData {
```

Comment:
Let's sort this type (and its `impl` block) below `Chunk`, since that's where it's used.

Reply:
This is what you were asking for, right?

Reply:
Yep, thanks!